libraries/AP_HAL_Linux/VideoIn.cpp

```diff
@@ -250,6 +250,42 @@ void VideoIn::prepare_capture()
     }
 }
 
+void VideoIn::shrink_8bpp(uint8_t *buffer, uint8_t *new_buffer,
+                          uint32_t width, uint32_t height, uint32_t left,
+                          uint32_t selection_width, uint32_t top,
+                          uint32_t selection_height, uint32_t fx, uint32_t fy)
+{
+    uint32_t i, j, k, px, block_x, block_y, block_position;
+    uint32_t out_width = selection_width / fx;
+    uint32_t out_height = selection_height / fy;
+    uint32_t width_per_fy = width * fy;
+    uint32_t fx_fy = fx + fy;
```

Comment:
and here I assume that since it is the dividing factor for a block, that it should be
<code>
fx_fy = fx*fy;